test/src/unit-status.cc

```diff
@@ -0,0 +1,45 @@
+#include <status.h>
+#include <catch.hpp>
+
+using namespace tiledb;
+
+TEST_CASE("Test Status ok") {
+  Status st = Status::Ok();
+  CHECK(st.ok());
+  st = Status::Error("err msg");
+  CHECK(!st.ok());
+}
+
+TEST_CASE("Test Status Code and Message") {
+  Status ok = Status::Ok();
+  CHECK(StatusCode::Ok == ok.code());
+
+  Status err = Status::Error("err msg");
+  CHECK(StatusCode::Error == err.code());
+  CHECK_THAT(err.message(), Catch::Equals("err msg"));
+}
+
+TEST_CASE("Test Status to_string") {
+  Status ok = Status::Ok();
+  CHECK_THAT(ok.to_string(), Catch::Equals("Ok"));
+
+  Status err = Status::Error("err msg");
+  CHECK_THAT(err.to_string(), Catch::Equals("Error: err msg"));
+}
+
+TEST_CASE("Test Status code_to_string") {
+  Status ok = Status::Ok();
+  CHECK_THAT(ok.code_to_string(), Catch::Equals("Ok"));
+
+  Status err = Status::Error("err message");
+  CHECK_THAT(err.code_to_string(), Catch::Equals("Error"));
+}
+
+TEST_CASE("Test Status posix_code") {
+  Status st = Status::Ok();
+  // check that posix code is < 0 by default
```

Comment:
-1 means that no code is set.  we don't use the posix code currently, I'm keeping it there so that we will be able to pass errno information (from gzip for example) that way we can map it to the right error message.